packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -2466,7 +2480,8 @@ export class StripeHelper extends StripeHelperBase {
       tax: invoiceTaxAmountInCents,
     } = invoice;
 
-    const nextInvoiceDate = subscriptionLineItem.period.end;
+    const nextInvoiceDate =
+      subscriptionLineItem?.period.end || invoiceitemLineItem?.period.end;
```

Comment:
Is it possible for an `invoiceitemLineItem` to not have a `period`? `nextInvoiceDate` was previously a value we could guarantee was defined.

Reply:
According to the Stripe types, period should always be available on any LineItem type.

Reply:
Can I suggest a small refactor for readability and so TS doesn't get confused about whether `invoiceitemLineItem` is `undefined`?

```diff
diff --git a/packages/fxa-auth-server/lib/payments/stripe.ts b/packages/fxa-auth-server/lib/payments/stripe.ts
index c6f86d9d54..bfac884370 100644
--- a/packages/fxa-auth-server/lib/payments/stripe.ts
+++ b/packages/fxa-auth-server/lib/payments/stripe.ts
@@ -2382,15 +2382,12 @@ export class StripeHelper extends StripeHelperBase {
 
     // In certain instances the invoice won't have a 'subscription' line item.
     // In those cases, select the 'invoiceitem' without proration_details.credited_items
-    const invoiceitemLineItem = !subscriptionLineItem
-      ? invoice.lines.data.find(
-          (line) =>
-            line.type === 'invoiceitem' &&
-            !line.proration_details?.credited_items
-        )
-      : undefined;
-
-    if (!subscriptionLineItem && !invoiceitemLineItem) {
+    const invoiceitemLineItem = invoice.lines.data.find(
+      (line) =>
+        line.type === 'invoiceitem' && !line.proration_details?.credited_items
+    );
+    const lineItem = subscriptionLineItem || invoiceitemLineItem;
+    if (!lineItem) {
       // No subscription or invoiceitem is present for the invoice. This should never happen
       // since all invoices have a related incoming subscription as one of the line items.
       throw error.internalValidationError(
@@ -2403,7 +2400,7 @@ export class StripeHelper extends StripeHelperBase {
     }
 
     // Dig up & expand objects in the invoice that usually come as just IDs
-    const plan = subscriptionLineItem?.plan || invoiceitemLineItem?.plan;
+    const plan = lineItem?.plan;
     if (!plan) {
       // No plan is present if this is not a subscription or proration, which
       // should never happen as we only have subscriptions.
@@ -2480,8 +2477,7 @@ export class StripeHelper extends StripeHelperBase {
       tax: invoiceTaxAmountInCents,
     } = invoice;
 
-    const nextInvoiceDate =
-      subscriptionLineItem?.period.end || invoiceitemLineItem?.period.end;
+    const nextInvoiceDate = lineItem.period.end;
 
     const invoiceDiscountAmountInCents =
       (invoice.total_discount_amounts &&
@@ -2530,7 +2526,7 @@ export class StripeHelper extends StripeHelperBase {
       invoiceDiscountAmountInCents,
       invoiceTaxAmountInCents,
       invoiceDate: new Date(invoiceDate * 1000),
-      nextInvoiceDate: nextInvoiceDate && new Date(nextInvoiceDate * 1000),
+      nextInvoiceDate: new Date(nextInvoiceDate * 1000),
       productId,
       productName,
       planId,
```